BUG: TdatFormatError Unrecognized data type in s390x job

Example log: https://github.com/astropy/astropy/actions/runs/14379962816/job/40321176784

Snippet:

```
 ___________________ test_roundtrip_masked[fmt_name_class32] ____________________  

  fmt_name_class = ('tdat', <class 'astropy.io.ascii.tdat.Tdat'>)
...
          try:
  >           ascii.write(t, out, format=fmt_name, fast_writer=fast)
  
astropy/io/ascii/tests/test_write.py:886: 
...
             else:
  >               raise TdatFormatError(
                      f'Unrecognized data type `{col.info.dtype}` for column "{col.info.name}".'
                  )
  E               astropy.io.ascii.tdat.TdatFormatError: Unrecognized data type `>U1` for column "c".
  E               See details in https://heasarc.gsfc.nasa.gov/docs/software/dbdocs/tdat.html
  
 astropy/io/ascii/tdat.py:527: TdatFormatError
```

Only happens on s390x job. Probably caused by:

* https://github.com/astropy/astropy/pull/16780

cc @d-giles @esabol @taldcroft @dhomeier

--- a/astropy/io/ascii/tdat.py
+++ b/astropy/io/ascii/tdat.py
@@ -515,19 +515,18 @@ class TdatHeader(basic.BasicHeader):
         lines.append("# Table Parameters")
         lines.append("#")
         for col in self.cols:
-            if str(col.info.dtype) in self._dtype_dict_out:
-                ctype = self._dtype_dict_out[str(col.info.dtype)]
-            elif "int" in str(col.info.dtype):
+            if str(col_type := col.info.dtype) in self._dtype_dict_out:
+                ctype = self._dtype_dict_out[str(col_type)]
+            elif col_type.kind == "i":
                 ctype = "int4"
-            elif "float" in str(col.info.dtype):
+            elif col_type.kind == "f":
                 ctype = "float8"
-            elif "<U" in str(col.info.dtype):
-                ctype = f"char{str(col.info.dtype).rsplit('<U', maxsplit=1)[-1]}"
+            elif col_type.kind == "U":
+                ctype = f"char{col_type.itemsize // 4}"
             else:
                 raise TdatFormatError(
-                    f'Unrecognized data type `{col.info.dtype}` for column "{col.info.name}".'
+                    f'Unrecognized data type `{col_type}` for column "{col.info.name}".'
                 )
-                # ctype = f"{col.info.dtype}"
             col_name = col.info.name
             if len(col_name) >= 24:
                 warn(

--- a/astropy/io/ascii/tests/test_tdat.py
+++ b/astropy/io/ascii/tests/test_tdat.py
@@ -188,7 +188,7 @@ def test_read_tdat():
         "Declination",
         "Empty",
     ]
-    dtypes = [np.int32, np.int32, "<U3", float, float, float]
+    dtypes = [np.int32, np.int32, np.dtype("U3"), float, float, float]
     units = [None, None, None, "deg", "deg", None]
     meta = [
         {"ucd": "meta.id", "index": "key"},
@@ -245,7 +245,7 @@ def test_full_table_content():
         ),
     }
     assert len(test_table) == 7
-    dtypes = [np.int32, np.int32, "<U3", float, float, float]
+    dtypes = [np.int32, np.int32, np.dtype("U3"), float, float, float]
     descriptions = [
         "Unique Identifier for Entry",
         "Source ID Number",

